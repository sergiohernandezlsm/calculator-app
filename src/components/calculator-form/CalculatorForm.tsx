import React from 'react';
import Form from 'react-bootstrap/Form';
import FormInput from '../form-input';
import { FieldTypes } from '../../types';
import styles from './CalculatorForm.module.scss';

export interface Props {
  calculatorHandler: (x: React.ChangeEvent<HTMLInputElement>) => void;
  formField: FieldTypes[];
}

const CalculatorForm: React.FC<Props> = (props) => {
  return (
    <div className={styles.formWrapper}>
      <Form>
        {props.formField.map((field: FieldTypes, index: number) => {
          return (
            <div key={`key-${index}`}>
              <FormInput
                calculatorHandler={props.calculatorHandler}
                key={`key-${index}`}
                placeholder={field.placeholder}
                name={field.name}
                text={field.text}
                label={field.label}
                min={field.min}
                max={field.max}
              />
            </div>
          )
        })}
      </Form>
    </div>
  )
};

export default CalculatorForm;
