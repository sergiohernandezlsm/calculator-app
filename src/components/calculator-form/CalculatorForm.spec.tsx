import { shallow } from 'enzyme';
import CalculatorForm from './CalculatorForm';

describe('CalculatorForm Component', () => {
  it('renders <CalculatorForm /> component', () => {
    const wrapper = shallow(<CalculatorForm />);
    expect(wrapper).toMatchSnapshot();
  });
});