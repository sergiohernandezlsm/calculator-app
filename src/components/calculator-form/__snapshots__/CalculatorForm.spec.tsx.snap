// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorForm Component renders <CalculatorForm /> component 1`] = `ShallowWrapper {}`;
